akka-http-core/src/test/scala/akka/http/impl/engine/client/HostConnectionPoolSpec.scala

```diff
@@ -404,6 +404,12 @@ class HostConnectionPoolSpec extends AkkaSpec(
         @volatile var shouldFail = true
         val connectionCounter = new AtomicInteger()
         override def onNewConnection(requestPublisher: Publisher[HttpRequest], responseSubscriber: Subscriber[HttpResponse]): Future[Http.OutgoingConnection] = {
+          while (connectionCounter == null) {
```

Comment:
Wait wat how can it be nul if it is a `val = böö new AtomicInteger`?

Reply:
I might be wrong, but my theory was somehow the `onNewConnection` callback was triggered from a field initialization in a superclass, which happens before the field initialization in this anonymous subclass. Something roundabout like:
```
@ abstract class A {
    val v = {
      runCallback();
      42;
    }
    def runCallback(): Unit
  }
defined class A

@ class B extends A {
    val b = 42
    override def runCallback(): Unit = {
      println(b)
    }
  }
defined class B

@ new B()
0
res2: B = ammonite.$sess.cmd1$B@5a7b309b
```

Reply:
(the stacktrace at https://jenkins.akka.io:8498/job/akka-http-nightly/334/AKKA_VERSION=default,Node=akka-http-nightly-node,SCALA_VERSION=2.12.8,jdk=AdoptOpenJDK%208/consoleText suggested this field was null when referenced inside the callback)

Reply:
Maybe a simpler fix is then to pull out those fields before `new SetupWithServerProbes`?

Reply:
I changed it so we can merge soon. It failed another time yesterday.